app/ui/src/main/java/com/fsck/k9/ui/K9Drawer.java

```diff
@@ -64,24 +82,68 @@ public K9Drawer(MessageList parent, Bundle savedInstanceState) {
+    private AccountHeader buildAccountHeader() {
+        AccountHeaderBuilder headerBuilder = new AccountHeaderBuilder()
+                .withActivity(parent)
+                //.withTextColorRes(R.color.primary_text_secondary_when_activated_material)
+                .withTranslucentStatusBar(false);
+
         if (!K9.isHideSpecialAccounts()) {
-            drawer.addItems(new PrimaryDrawerItem()
-                            .withName(R.string.integrated_inbox_title)
-                            .withIcon(getResId(R.attr.iconUnifiedInbox))
-                            .withIdentifier(DRAWER_ID_UNIFIED_INBOX),
-                    new DividerDrawerItem());
+            headerBuilder.addProfiles(new ProfileDrawerItem()
+                    .withNameShown(true)
+                    .withName(R.string.integrated_inbox_title)
+                    .withEmail(parent.getString(R.string.integrated_inbox_detail))
+                    .withIcon(new IconicsDrawable(parent, FontAwesome.Icon.faw_users)
+                            .colorRes(R.color.material_drawer_background).backgroundColor(Color.GRAY)
+                            .sizeDp(56).paddingDp(8))
+                    .withSetSelected(unifiedInboxSelected)
+                    .withIdentifier(DRAWER_ID_UNIFIED_INBOX)
+            );
+        }
 
-            headerItemCount += 2;
+        List <Account> accounts = preferences.getAccounts();
+        for (int i = preferences.getAccounts().size() - 1; i >= 0; i--) {
+            Account account = accounts.get(i);
+            long drawerId = (account.getAccountNumber()+1) << DRAWER_ACCOUNT_SHIFT;
+
+            headerBuilder.addProfiles(new ProfileDrawerItem()
+                    .withNameShown(true)
+                    .withName(account.getDescription())
+                    .withEmail(account.getEmail())
+                    .withIcon(new IconicsDrawable(parent, FontAwesome.Icon.faw_user_alt)
+                            .colorRes(R.color.material_drawer_background).backgroundColor(account.getChipColor())
+                            .sizeDp(56).paddingDp(14))
+                    .withIdentifier(drawerId)
+                    .withSetSelected(false)
+                    .withTag(account)
+            );
         }
+
+        accountHeader = headerBuilder
+                .withOnAccountHeaderListener(new AccountHeader.OnAccountHeaderListener() {
+                    @Override
+                    public boolean onProfileChanged(View view, IProfile profile, boolean currentProfile) {
+                        if (profile.getIdentifier() == DRAWER_ID_UNIFIED_INBOX) {
+                            parent.openUnifiedInbox();
+                            return false;
+                        } else {
+                            Account account = (Account) ((ProfileDrawerItem) profile).getTag();
+                            updateUserAccountsAndFolders(account);
+                            Accounts.openRealAccount(account, parent);
```

Comment:
`Accounts` will be removed once we have the account switcher in `MessageList` (and all the functionality reachable from `Accounts` is accessible from elsewhere). So the functionality to switch accounts should live in `MessageList`.

Reply:
I'm groping a bit in the dark here, it would be great if you could take over this, or we can wait until `Accounts` is settled.

Reply:
done